march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator_view.py

```diff
@@ -0,0 +1,231 @@
+import math
+import os
+import subprocess
+
+import pyqtgraph as pg
+from pyqtgraph.Qt import QtCore
+from python_qt_binding import loadUi
+from python_qt_binding.QtWidgets import (QCheckBox, QComboBox, QDoubleSpinBox,
+                                         QFileDialog, QFrame, QHeaderView,
+                                         QLineEdit, QMessageBox, QPushButton,
+                                         QSlider, QSpinBox, QWidget)
+from qt_gui.plugin import Plugin
+import rospkg
+import rospy
+import rviz
+from sensor_msgs.msg import JointState
+from tf import (ConnectivityException, ExtrapolationException, LookupException,
+                TransformListener)
+
+from .gait_generator_controller import GaitGeneratorController
+from .joint_plot import JointPlot
+from .joint_table_controller import JointTableController
+from .time_slider_thread import TimeSliderThread
+
+
+class GaitGeneratorView(Plugin):
+    def __init__(self, context):
+        super(GaitGeneratorView, self).__init__(context)
+
+        self.joint_widgets = {}
+        self.tf_listener = TransformListener()
+        self.joint_state_pub = rospy.Publisher('joint_states', JointState, queue_size=10)
+
+        self.build_ui(context)
+
+        self.controller = GaitGeneratorController(self)
+
+    # Called by __init__
+    def build_ui(self, context):
+        # Start UI construction
+        self._widget = QWidget()
```

Comment:
I think it would be nice to split the `Widget` from the `Plugin`, we also do this in `march_rqt_note_taker`. You essentially subclass from `QWidget`, so that you can implement it directly instead of accessing and modifying the object over and over. It think it also creates a nice separation of the actual plugin, which creates the models, controllers and view, and the GUI. I took the inspiration from [`rqt_console`](https://github.com/ros-visualization/rqt_console)